src/main/java/teamroots/embers/upgrade/UpgradeMiniBoiler.java

```diff
@@ -32,23 +29,7 @@ public int getLimit(TileEntity tile) {
     @Override
     public void throwEvent(TileEntity tile, UpgradeEvent event) {
         if(event instanceof EmberEvent) {
-            setHeat(((EmberEvent) event).getAmount());
-        }
-    }
-
-    public void setHeat(double heat) {
-        this.heat = heat;
-        this.active = true;
-    }
-
-    @Override
-    public boolean doWork(TileEntity tile, List<IUpgradeProvider> upgrades) {
-        if(active) {
-            if (this.tile instanceof TileEntityMiniBoiler)
-                ((TileEntityMiniBoiler) this.tile).boil(heat);
-            active = false;
+            ((TileEntityMiniBoiler) this.tile).boil(((EmberEvent) event).getAmount());
```

Comment:
i don't approve of this. the code was written like that for a reason. the ember event can be thrown without any work being done after all, so the heat value is stashed until we actually do work.

Reply:
and what's the point in that? As it is right now, it's super counter intuitive and feels like a bug, from the player perspective, at least. Moreover, that heat doesn't get serialized and saved, which means that there's always waste of heat, no matter what. Is that intended as well?

Reply:
how are you noticing this as a player though, it's a purely wetworks change to prevent "dupes"? I also don't really consider one tick of no steam with no fuel consumption either when you load the chunk/world to be loss.

Reply:
I'm making a modpack where ember shards and such are rare and the player needs to make the most efficient use of it, which includes having ember dial on the Pressure Refinery, for example, and the fluid dial on the mini boiler, to have precise control of their production. Thing is, the way you have implemented the mini boiler, it's VERY easy to explode the damn thing, you have to input fuel into the Pressure Refinery very slowly otherwise the boiler WILL explode. 

Do you think I have changed this piece of code out of nowhere? I have changed it exactly because I have been having problems with the feedback from the boiler and went looking into the code to figure out where the hell it behaves so stupidly. Here's how it went for me, as a player:

"Hmm, I've used an ember shard, it produces ember energy but no steam whatsoever... Did I do something wrong? Am I supposed to put the boiler connected to another part of the machine? Maybe this one doesn't accept the boiler upgrade? Lemme try putting another shard in... it works now! Is it a bug? Did it void the heat from the first shard? *make changes to CraftTweaker scripts, reboot the game* No steam yet again?! What the hell?!"

And you're trying to tell me this is intended...

Reply:
@Kurtchekov any chance that you still have that pack so i can have a go at it?

i tried searching in case i could luck out and find it on cf before asking here

Reply:
> @Kurtchekov any chance that you still have that pack so i can have a go at it?
> 
> i tried searching in case i could luck out and find it on cf before asking here

I'm sorry, but I ended up postponing that project because multiple mods refused to make the changes that were needed for that degree of balance. I had to remove Embers a long time ago, I took over the coding for Immersive Tech (see Immersive Technology) and I'm still waiting for a few other mods to update, such as the overhauled NuclearCraft.